skllm/models/gpt_zero_shot_clf.py

```diff
@@ -152,6 +233,19 @@ def _extract_labels(self, y) -> List[str]:
     def _get_prompt(self, x) -> str:
         return build_zero_shot_prompt_mlc(x, repr(self.classes_), self.max_labels)
 
+    def _get_default_label(self):
+        """ Returns the default label based on the default_label argument. """
+        result = []
+        if isinstance(self.default_label, str) and self.default_label == "Random":
+            for cls, probability in zip(self.classes_, self.probabilities_):
+                coin_flip = random.choices([0,1], [1-probability, probability])[0]
+                if coin_flip == 1:
+                    result.append(cls)
+        else:
+            result = self.default_label
```

Comment:
I think that this behaviour might still be a bit confusing for the users. If a string != "Random" is provided instead of a list, the label will again be a string. So, I would still add an additional type check and convert to list whenever applicable.

If you intentionally want to have a flexibility of having non-list outputs, maybe this could be done for `default_label = None` as a special case (this should be properly documented then). But in my opinion, it would be always better to have a list as an output.

Reply:
See lines 221-222, The output can be either a list or a None.
They main point of this PR is to facilitate a way to ignore the model predictions when he fails,
that is most commonly achieved by setting label = None

Reply:
@Nadav-Barak you mean lines 210-211? I must have missed that change.